includes/class.llms.achievement.php

```diff
@@ -124,8 +124,16 @@ public function __call( $name, $args ) {
 		}
 	}
 
+	/**
+	 * Constructor.
+	 *
+	 * @since Unknown.
+	 * @deprecated [version] `LLMS_Achievement::__construct()` is deprecated with no replacement.
+	 */
 	public function __construct() {
 
+		_deprecated_function( __METHOD__, '[version]' );
```

Comment:
I think the idea here is that the whole class is deprecated so adding a deprecation notice to the constructor alerts developers. In which case I think we should add a message and instead of outputting that the constructor is deprecated, we should output that the class is deprecated
```suggestion
		_deprecated_function( __CLASS__, '[version]' );
```

Reply:
Interesting! I didn't try just the class name. Initially I did this.
```php
// Log "The LLMS_Achievement class is deprecated since version [version] with no alternative available.".
_deprecated_function( 'The ' . __CLASS__ . ' class', '[version]' );
```
However, when I added `@expectedDeprecated LLMS_Certificate::__construct` to [`LLMS_Test_LLMS_User_Certificate::test_get_merge_data_deprecated_hook()`](https://github.com/gocodebox/lifterlms/blob/1931-refactor-engagement-sync/tests/phpunit/unit-tests/models/class-llms-test-model-llms-user-certificate.php#L648) I got errors.

Providing just the class name seem to work very well.